project-base/app/src/Model/Customer/User/RegistrationDataFactory.php

```diff
@@ -61,4 +64,25 @@ public function createWithArgument(Argument $argument): RegistrationData
 
         return $registrationData;
     }
+
+    /**
+     * @param \Hybridauth\User\Profile $profile
+     * @return \App\Model\Customer\User\RegistrationData
+     */
+    public function createFromSocialNetworkProfile(Profile $profile): RegistrationData
+    {
+        $registrationData = $this->createForDomainId($this->domain->getId());
+        $countries = $this->countryFacade->getAllEnabledOnCurrentDomain();
+
+        $registrationData->firstName = $profile->firstName ?? '';
+        $registrationData->lastName = $profile->lastName ?? '';
+        $registrationData->email = $profile->email;
+        $registrationData->street = '';
+        $registrationData->city = '';
+        $registrationData->postcode = '';
+        $registrationData->country = $countries[0];
+        $registrationData->password = $this->hashGenerator->generateHashWithoutConfusingCharacters(CustomerUserPasswordFacade::MINIMUM_PASSWORD_LENGTH);
```

Comment:
When the password policy is changed on a project (e.g. special characters are required), this implementation might be a problem, right? Even at this moment, it feels like it might lead to quite weak passwords :thinking: Let's discuss it, please

Reply:
fixed in https://github.com/shopsys/shopsys/pull/3154/commits/68ae66fd9ef6fd6cfcdaf20831bd36b8e38f3186